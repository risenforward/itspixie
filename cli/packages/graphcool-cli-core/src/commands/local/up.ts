import { Command, flags, Flags, AuthenticateCustomerPayload } from 'graphcool-cli-engine'
import Docker from './Docker'
import * as chalk from 'chalk'
const debug = require('debug')('up')

export default class Up extends Command {
  static topic = 'local'
  static command = 'up'
  static description = 'Start local development cluster (Docker required)'
  static group = 'local'
  static flags: Flags = {
    name: flags.string({
      char: 'n',
      description: 'Name of the new instance',
      defaultValue: 'local'
    }),
  }
  async run() {
    const {name} = this.flags

    const docker = new Docker(this.out, this.config, this.env, name)

    const {envVars: {MASTER_TOKEN, PORT}} = await docker.up()

    this.out.log('')
    this.out.action.start('Waiting for Graphcool to initialize. This can take several minutes')
    const cluster = (this.env.rc.clusters && this.env.rc.clusters[name]) ? this.env.rc.clusters[name] : null
    const host = (cluster && typeof cluster !== 'string') ? cluster.host : 'http://localhost:' + PORT
    const endpoint = host + '/system'
    await this.client.waitForLocalDocker(endpoint)

    const {token}: AuthenticateCustomerPayload = await this.client.authenticateCustomer(endpoint, MASTER_TOKEN)


    if (!cluster) {
      debug('Setting cluster')
      this.env.setGlobalCluster(name, {
        host,
        token,
      })
      this.env.saveGlobalRC()
    }

    this.out.action.stop()

    if (!cluster) {
      this.out.log(`\nSuccess! Added local cluster ${chalk.bold(`\`${name}\``)} to ${this.config.globalRCPath}\n`)
    } else {
      this.out.log(`\nSuccess! Cluster ${name} already exists and is up-to-date.`)
    }
    this.out.log(`To get started, execute
    
  ${chalk.green('$ graphcool init')}
  ${chalk.green('$ graphcool deploy')}
`)
  }
}
