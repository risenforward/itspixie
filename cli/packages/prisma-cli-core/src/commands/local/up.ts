import {
  Command,
  flags,
  Flags,
  AuthenticateCustomerPayload,
} from 'prisma-cli-engine'
import { Cluster } from 'prisma-yml'
import Docker from './Docker'
import chalk from 'chalk'
import { prettyTime } from '../../util'
const debug = require('debug')('up')

export default class Up extends Command {
  static topic = 'local'
  static command = 'start'
  static description = 'Start local development cluster'
  static group = 'local'
  static flags: Flags = {
    name: flags.string({
      char: 'n',
      description: 'Name of the cluster instance',
      defaultValue: 'local',
    }),
  }
  async run() {
    const { name } = this.flags

    const docker = new Docker(this.out, this.config, this.env, name)

    this.out.action.start('Booting local development cluster')

    await docker.init()

    const cluster = docker.saveCluster()

    await docker.up()

    const before = Date.now()
    await this.client.waitForLocalDocker(cluster.getDeployEndpoint())

    this.out.action.stop(prettyTime(Date.now() - before))

    const showInit = !this.definition.definition
    //     this.out.log(`To get started, execute

    // ${showInit && `  ${chalk.green('$ prisma init')}\n`}  ${chalk.green(
    //       '$ prisma deploy',
    //     )}
    // `)
  }
}
