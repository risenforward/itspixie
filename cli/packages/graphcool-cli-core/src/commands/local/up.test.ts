import * as nock from 'nock'
import Up from './up'
import up from './nocks/up'

jest.mock('./Docker')
const Docker = require('./Docker').default
Docker.mockImplementation(() => {
  return {
    up: () => {
      return Promise.resolve({envVars: {MASTER_TOKEN: 'token', PORT: '60000'}})
    }
  }
})

afterAll(() => {
  nock.cleanAll()
})

describe('up', () => {
  test('in empty dir', async () => {
    up()
    const result = await Up.mock()
    expect(result.out.stdout.output).toMatchSnapshot()
  })
})
