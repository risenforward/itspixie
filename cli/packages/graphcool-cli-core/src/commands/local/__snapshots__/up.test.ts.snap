// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`up in empty dir 1`] = `
"

Success! Added local cluster \`local\` to /var/folders/lc/rzmq7fnn4_jbrcsbz_f1kyfr0000gr/T/cj8mu2b170001bkyvp1vqdpig/.graphcoolrc

To get started, execute
    
  $ graphcool init
  $ graphcool deploy

"
`;
